src/MUIDataTable.js

```diff
@@ -677,14 +670,15 @@ class MUIDataTable extends React.Component {
     return transformedData;
   };
 
-  setTableData(props, status, dataUpdated, callback = () => {}) {
+  setTableData(props, status, dataUpdated, callback = () => {}, fromConstructor = false) {
```

Comment:
It doesn't seem necessary to add fromConstructor.
Because it can be judged by the existence of callback.

And I think it's better to separate the function.

I think it has too heavy functions.

Reply:
The constructor sets a default function for that callback. I also wanted to make it explicit at what was happening, since the constructor is the only place where the unique behavior happens.

Reply:
>The constructor sets a default function for that callback. I also wanted to make it explicit at what was happening, since the constructor is the only place where the unique behavior happens.

Oh, I see. There is a default. hmmm...